test/data/test_data.py

```diff
@@ -264,3 +266,25 @@ def test_basic_feature_store():
     assert 'x' in data.__dict__['_store']
     data.remove_tensor(attr_name='x', index=None)
     assert 'x' not in data.__dict__['_store']
+
+
+# Materialized Graph ##########################################################
```

Comment:
I personally find the naming of `MaterializedGraph` somewhat confusing - it is not clear why we do not just simply name it `Graph` or `GraphStore`.